server/http_handler.go

```diff
@@ -1249,3 +1258,88 @@ func (h *mvccTxnHandler) handleMvccGetByTxn(params map[string]string) (interface
 	}
 	return h.getMvccByStartTs(uint64(startTS), startKey, endKey)
 }
+
+// ReportServerInfo is only used to report the servers info when do http request
+type ReportServerInfo struct {
+	IsOwner         bool                `json:"is_owner"`
+	SelfServerInfo  *util.DDLServerInfo `json:"self_server_info"`
+	OwnerServerInfo *util.DDLServerInfo `json:"owner_server_info,omitempty"`
+}
+
+// ServeHTTP handles request of ddl server info.
+func (h ddlServerInfoHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	se, err := session.CreateSession(h.store.(kv.Storage))
+	if err != nil {
+		writeError(w, errors.New("create session error"))
+		return
+	}
+	ddl := domain.GetDomain(se.(sessionctx.Context)).DDL()
```

Comment:
There is no need to create a session to get a domain, just use `session.GetDomain` is ok.